src/app/node-data/component.ts

```diff
@@ -191,35 +191,44 @@ export class NodeDataComponent extends BaseFormValidator implements OnInit, OnDe
   }
 
   isOperatingSystemSupported(os: OperatingSystem): boolean {
-    // If VSphere is selected enable OS only if it is also defined in the datacenter spec
-    if (this._hasSystemTemplate(NodeProvider.VSPHERE, os)) {
-      return true;
-    }
+    if (this._datacenterSpec.spec.enabledOperatingSystems.length > 0) {
+      for (const enabledOS of this._datacenterSpec.spec.enabledOperatingSystems) {
+        if (enabledOS as OperatingSystem == os ) {
```

Comment:
Use `===` instead of `==`. Run the checks like `npm run check` locally to detects issues like this automatically.